polkadot/node/subsystem-types/src/messages.rs

```diff
@@ -1106,6 +1110,9 @@ pub struct ProspectiveValidationDataRequest {
 	pub candidate_relay_parent: Hash,
 	/// The parent head-data hash.
 	pub parent_head_data_hash: Hash,
+	/// Optionally, the head-data of the parent.
+	/// This will be provided for collations with elastic scaling enabled.
+	pub maybe_parent_head_data: Option<HeadData>,
```

Comment:
we could use an enum with 2 variants instead of adding an option: `ParentHeadDataHash` and `ParentHeadData`.